tailscale/resource_dns_configuration_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) David Bond, Tailscale Inc, & Contributors
+// SPDX-License-Identifier: MIT
+
+package tailscale
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"tailscale.com/client/tailscale/v2"
+)
+
+const testDNSConfigurationCreate = `
+	resource "tailscale_dns_configuration" "test_configuration" {
+		nameservers {
+			address            = "8.8.8.8"
+		}
+		nameservers {
+			address            = "1.1.1.1"
+			use_with_exit_node = true
+		}
+		split_dns {
+			domain             = "foo.example.com"
+			nameservers {
+				address            = "1.1.1.2"
+				use_with_exit_node = true
+			}
+			nameservers {
+				address            = "1.1.1.3"
+			}
+		}
+		split_dns {
+			domain             = "bar.example.com"
+			nameservers {
+				address            = "8.8.8.2"
+				use_with_exit_node = true
+			}
+		}
+		search_paths       = ["example.com", "anotherexample.com"]
+		override_local_dns = true
+		magic_dns = true
+	}`
+
+const testDNSConfigurationUpdate = `
+	resource "tailscale_dns_configuration" "test_configuration" {
+		nameservers {
+			address            = "8.8.8.8"
+			use_with_exit_node = true
+		}
+		split_dns {
+			domain             = "bar.example.com"
+			nameservers {
+				address            = "8.8.8.2"
+				use_with_exit_node = false
+			}
+		}
+		search_paths       = ["anotherexample.com"]
+		override_local_dns = false
+		magic_dns = false
+	}`
+
+func TestProvider_TailscaleDNSConfiguration(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		IsUnitTest: true,
+		PreCheck: func() {
+			testServer.ResponseCode = http.StatusOK
+			testServer.ResponseBody = nil
+		},
+		ProviderFactories: testProviderFactories(t),
+		Steps: []resource.TestStep{
+			testResourceCreated("tailscale_dns_configuration.test_configuration", testDNSConfigurationCreate),
+			testResourceDestroyed("tailscale_dns_configuration.test_configuration", testDNSConfigurationCreate),
+		},
+	})
+}
+
+func TestAccTailscaleDNSConfiguration(t *testing.T) {
+	const resourceName = "tailscale_dns_configuration.test_configuration"
+
+	checkProperties := func(expected *tailscale.DNSConfiguration) func(client *tailscale.Client, rs *terraform.ResourceState) error {
+		return func(client *tailscale.Client, rs *terraform.ResourceState) error {
+			actual, err := client.DNS().Configuration(context.Background())
+			if err != nil {
+				return err
+			}
+
+			if err := assertEqual(expected, actual, "wrong DNS configuration"); err != nil {
+				return err
+			}
+
+			return nil
+		}
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(t),
+		CheckDestroy: checkResourceDestroyed(resourceName, checkProperties(&tailscale.DNSConfiguration{
+			// TODO: we need empty values here because the server currently doesn't `omitempty`.
```

Comment:
Reminder to adjust / rewrite when we `omitempty` on the server

Reply:
Done.